.github/workflows/ci.yml

```diff
@@ -80,61 +123,74 @@ jobs:
         run: |
           ci-scripts/validate-lint.sh
   b2c_e2e_tests:
-    needs: no_retries
+    needs: [no_retries, validate_e2e_execution]
     name: E2E tests for B2C
     runs-on: ubuntu-latest
     strategy:
       matrix:
         containers: [1, 2, 3, 4, 5]
     steps:
       - uses: actions/checkout@v2
+      - name: Cache node_modules
+        id: cache-node-modules
+        uses: actions/cache@v2
+        with:
+          path: | 
+            node_modules
+            projects/storefrontapp-e2e-cypress/node_modules
+            ~/.cache/Cypress
+          key: nodemodules-${{ github.event.pull_request.base.sha }}
+          restore-keys: nodemodules-${{ github.event.pull_request.base.sha }}
       - name: Run e2es
         env: 
           SPA_ENV: ci,b2c
         run: |
           ci-scripts/e2e-cypress.sh
   b2c_ssr_e2e_tests:
-    needs: no_retries
+    needs: [no_retries, validate_e2e_execution]
     name: E2E tests for SSR B2C
```

Comment:
should we add the environment variable here? SPA_ENV: ci,b2c

Reply:
yea I guess we can, but did you determine if it was a mis-config / design from travis?